src/SAWScript/AST.hs

```diff
@@ -346,71 +350,71 @@ pShow :: PrettyPrint a => a -> String
 pShow = show . pretty 0
 
 class PrettyPrint p where
```

Comment:
Oh boy, this is confusing.  Different class name, same method, but with different arguments.  I don't know if it's constrained to this file or not, but it seems like a little alpha renaming could avoid some confusion (like `class SAWPretty p where pprint :: Int -> p -> PP.Doc ann` or something).  Again, not a situation created by this PR, but especially with the method aliasing this feels like we could avoid some future pain by changing this; it's up to you if you think that should be addressed in this PR.

Reply:
Yeah, probably `ppPrec` would be a better name; that's a convention that's also used within cryptol. Taking a larger view, a typeclass for prettyprinting with precedences might be something that should be pulled out into a separate, more widely reusable package. I'll open a ticket for this instead of tackling it in this PR.

Reply:
The class is exported, but it looks like it's not used anywhere but in this file. I still think it's worth opening a ticket, as I'd definitely consider this to be tech debt.

Reply:
See #950.